smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/waiters/JMESPathVisitor.kt

```diff
@@ -0,0 +1,412 @@
+                    .id("smithy.swift.synthetic#mappedCollection\$member")
+                    .target(transformedVar!!.shape)
+                    .build()
+                val returnType = ListShape.builder()
+                    .id("smithy.swift.synthetic#mappedCollection")
+                    .member(returnMember)
+                    .build()
+                return Variable(outerName, left.isOptional, returnType)
+            }
+            else -> throw Exception("Mapping a non-collection shape: ${left.shape}")
+        }
+    }
+
+    // Accesses a field on the parent variable & returns a variable containing the
+    // resulting expression.
+    private fun subfield(expression: FieldExpression, parentVar: Variable): Variable {
+        when (parentVar.shape) {
+            is StructureShape -> {
+                val parentMembers = parentVar.shape.members()
+                val subfieldMember = parentMembers.first { it.memberName == expression.name }
+                val subfieldShape = model.expectShape(subfieldMember.target)
+                val subfieldName = symbolProvider.toMemberName(subfieldMember)
+                val fieldOperator = "?.".takeIf { parentVar.isOptional } ?: "."
+                //
+                // Because all fields on Swift models are currently optional, every subfield value will also be
+                // optional.  Later we may have to replace this with logic to determine actual optionality,
+                // so:
+                // val subfieldIsOptional = !subfieldMember.hasTrait<RequiredTrait>()
+                // is temporarily replaced with:
+                val subfieldIsOptional = true
+
+                val subfieldVar = Variable(subfieldName, subfieldIsOptional, subfieldShape)
+                return addTempVar(subfieldVar, "\$L\$L\$L", parentVar.name, fieldOperator, subfieldName)
+            }
+            else -> {
+                throw Exception("Accessed subfield on parent: $parentVar")
+            }
+        }
+    }
+
+    // Performs a Boolean "and" of the left & right expressions
+    // A Swift compile error will result if both left & right aren't Booleans.
+    override fun visitAnd(expression: AndExpression): Variable {
+        val leftExp = expression.left!!.accept(this)
+        val rightExp = expression.right!!.accept(this)
+        val andResultVar = Variable("andResult", false, boolShape)
+        return addTempVar(andResultVar, "\$L && \$L", leftExp, rightExp)
+    }
+
+    // Perform a comparison of two values.
+    // The JMESValue type is used to provide conversion and comparison as needed between types
+    // that aren't comparable in "pure Swift" (i.e. Int to Double or String to RawRepresentable
+    // by String.)
+    // The Smithy comparator is a string that just happens to match up with all Swift comparators,
+    // so it is rendered into Swift as-is.
+    override fun visitComparator(expression: ComparatorExpression): Variable {
```

Comment:
It appears that we only support two comparators currently `==` and `<`. Can we perform some validation of `expression.comparator` to verify it's one that we support, this way we'll get an error during codegen if a new comparator is added that we don't support. imo that is preferable to getting an error during swift compile time as it makes it much more clear what the actual issue is.

Reply:
If you implement `==` and `<` in Swift, you support all the other Smithy operators because Swift provides all the others through default implementations on the `Equatable` and `Comparable` protocols.

Reply:
For reference here are Smithy's comparators: https://github.com/awslabs/smithy/blob/bf9fc069338fe78ed3dfc1f51820a0101dfec12b/smithy-jmespath/src/main/java/software/amazon/smithy/jmespath/ast/ComparatorType.java#L21

Reply:
Ahh that makes sense, i missed the `Comparable` conformance. 
awesome!